scripts/reports/stale-cases.ts

```diff
@@ -0,0 +1,167 @@
+// usage:
+// npx ts-node --transpile-only scripts/reports/stale-cases.ts
+
+import { CASE_STATUS_TYPES } from '@shared/business/entities/EntityConstants';
+import {
+  ServerApplicationContext,
+  createApplicationContext,
+} from '@web-api/applicationContext';
+import { appendFileSync } from 'fs';
+import {
+  calculateDifferenceInDays,
+  createISODateString,
+} from '@shared/business/utilities/DateHandler';
+import {
+  search,
+  searchAll,
+} from '@web-api/persistence/elasticsearch/searchClient';
+import PQueue from 'p-queue';
+
+const todayISO = createISODateString();
+const OUTPUT_DIR = `${process.env.HOME}/Documents`;
+const OUTPUT_FILENAME = `${OUTPUT_DIR}/stale-cases_${todayISO.split('T')[0]}.csv`;
+const CHUNK_SIZE = 50;
+const staleCases = {};
+
+const excludedCaseStatuses = [
+  CASE_STATUS_TYPES.closed,
+  CASE_STATUS_TYPES.closedDismissed,
+  CASE_STATUS_TYPES.onAppeal,
+];
+
+const getAllCasesNotInExcludedStatus = async ({
+  applicationContext,
+}: {
+  applicationContext: ServerApplicationContext;
+}): Promise<RawCase[]> => {
+  const { results } = await searchAll({
+    applicationContext,
+    searchParameters: {
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'entityName.S': 'Case',
+                },
+              },
+            ],
+            must_not: [
+              {
+                terms: {
+                  'status.S': excludedCaseStatuses,
+                },
+              },
+            ],
+          },
+        },
+        sort: [{ 'sortableDocketNumber.N': 'asc' }],
+      },
+      index: 'efcms-case',
+    },
+  });
+  return results;
+};
+
+const getMostRecentDocketEntry = async ({
+  applicationContext,
+  docketNumber,
+}: {
+  applicationContext: ServerApplicationContext;
+  docketNumber: string;
+}): Promise<RawDocketEntry | undefined> => {
+  const { results } = await search({
+    applicationContext,
+    searchParameters: {
+      body: {
+        from: 0,
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'entityName.S': 'DocketEntry',
+                },
+              },
+              {
+                term: {
+                  'docketNumber.S': docketNumber,
+                },
+              },
+            ],
+          },
+        },
+        size: 1,
+        sort: [{ 'receivedAt.S': 'desc' }],
+      },
+      index: 'efcms-docket-entry',
+    },
+  });
+  return results[0];
+};
+
+const isCaseStale = async ({
+  aCase,
+  applicationContext,
+}: {
+  aCase: RawCase;
+  applicationContext: ServerApplicationContext;
+}): Promise<void> => {
+  const mostRecentDocketEntry = await getMostRecentDocketEntry({
+    applicationContext,
+    docketNumber: aCase.docketNumber,
+  });
+  const deRcvdAt = mostRecentDocketEntry?.receivedAt;
+  const deAge = deRcvdAt ? calculateDifferenceInDays(todayISO, deRcvdAt) : 0;
+  if (deAge > 364) {
```

Comment:
You have awakened the *Constant Curmudgeon*, who will provide little substantive feedback and will instead whine about replacing magic numbers with constants.

Reply:
> You have awakened the Constant Curmudgeon, who will provide little substantive feedback and will instead whine about replacing magic numbers with constants.

This feels like it should be the text in a card game.

Reply:
Ha! Your suggestion is appreciated. :)